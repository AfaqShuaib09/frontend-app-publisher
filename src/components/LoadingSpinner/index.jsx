import React from 'react';
import { Icon } from '@edx/paragon';

const LoadingSpinner = () => (
  <div className="mx-auto text-center">
    <Icon
      id="spinner"
      className={['fa', 'fa-circle-o-notch', 'fa-spin', 'fa-3x', 'fa-fw']}
    />
  </div>
);

export default LoadingSpinner;
