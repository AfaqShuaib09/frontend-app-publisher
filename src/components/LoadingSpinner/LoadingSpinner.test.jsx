import React from 'react';
import { shallow } from 'enzyme';

import LoadingSpinner from './index';

describe('LoadingSpinner', () => {
  it('shows a loading spinner', () => {
    const component = shallow(<LoadingSpinner />);
    expect(component).toMatchSnapshot();
  });
});
