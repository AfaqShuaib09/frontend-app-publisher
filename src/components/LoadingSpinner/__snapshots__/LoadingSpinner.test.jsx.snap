// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner shows a loading spinner 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingSpinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <u
        className={
          Array [
            "fa",
            "fa-circle-o-notch",
            "fa-spin",
            "fa-3x",
            "fa-fw",
          ]
        }
        hidden={true}
        id="spinner"
      />,
      "className": "mx-auto text-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": Array [
          "fa",
          "fa-circle-o-notch",
          "fa-spin",
          "fa-3x",
          "fa-fw",
        ],
        "hidden": true,
        "id": "spinner",
        "screenReaderText": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <u
          className={
            Array [
              "fa",
              "fa-circle-o-notch",
              "fa-spin",
              "fa-3x",
              "fa-fw",
            ]
          }
          hidden={true}
          id="spinner"
        />,
        "className": "mx-auto text-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": Array [
            "fa",
            "fa-circle-o-notch",
            "fa-spin",
            "fa-3x",
            "fa-fw",
          ],
          "hidden": true,
          "id": "spinner",
          "screenReaderText": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
